SDK-v4/gdpr/app/src/main/java/com/localytics/gdpr/LocalyticsGDPRWrapper.java

```diff
@@ -0,0 +1,246 @@
+                }
+            });
+        } else {
+            Localytics.closeSession();
+        }
+    }
+
+    public void onActivityResume(final Activity activity) {
+        if (optedOut) {
+            return;
+        }
+
+        if (queueDatapoints) {
+            eventQueue.add(new Runnable() {
+                @Override
+                public void run() {
+                    Localytics.onActivityResume(activity);
+                }
+            });
+        } else {
+            Localytics.onActivityResume(activity);
+        }
+    }
+
+    public void onActivityPause(final Activity activity) {
+        if (optedOut) {
+            return;
+        }
+
+        if (queueDatapoints) {
+            eventQueue.add(new Runnable() {
+                @Override
+                public void run() {
+                    Localytics.onActivityPause(activity);
+                }
+            });
+        } else {
+            Localytics.onActivityPause(activity);
+        }
+    }
+
+    public void pauseDataUploading(boolean pause) {
+        queueDatapoints = pause;
+        if (!pause) {
+            for (Runnable event : eventQueue) {
+                event.run();
+            }
+        }
+    }
+
+    public void setLocationMonitoringEnabled(boolean locationMonitoringEnabled) {
+        Localytics.setLocationMonitoringEnabled(locationMonitoringEnabled);
+    }
+
+    //    Getters
+
+    public String getCustomerId() {
+        return Localytics.getCustomerId();
+    }
+
+    // User login, logout, opt in, opt out
+
+    public void setPrivacyOptedOut(boolean privacyOptedOut) {
+        if (optedOut == privacyOptedOut) {
+            return;
+        }
+        updateOptedOutState(privacyOptedOut);
+    }
+
+    public void logIn(String customerId, boolean optedOut) {
+        Localytics.setCustomerId(customerId);
+        updateOptedOutState(optedOut);
+    }
+
+    public void logout() {
+        Localytics.setCustomerId(null);
+        updateOptedOutState(false);
+    }
+
+    private void updateOptedOutState(boolean optedOut) {
+        this.optedOut = optedOut;
+        Localytics.setProfileAttribute("privacy_delete", optedOut ? 1 : 0);
+        Localytics.setOptedOut(optedOut);
+        if (optedOut) {
+            //Make sure to delete all identifiers that you have set.
+            Localytics.setIdentifier("your defined identifiers", null);
+//            If you are using places, we suggest shutting it down when a user opts out.
+//            Localytics.setLocationMonitoringEnabled(false);
+            eventQueue.clear();
+        } else {
+//            If you are using places, make sure to re-enable places when a user opts back in.
+//            Localytics.setLocationMonitoringEnabled(true);
+        }
+        Localytics.upload();
+    }
+
+    // Example of implementing tagEvent and setProfileAttribute.
+    // Any additional event or profile methods overridden should follow a similar protocol.
```

Comment:
Should we list more APIs that should be overridden? (i.e `tagScreen` or any standard events)

Reply:
There are a lot... I just listed a few as a basic idea, is it worth the effort we feel?

Reply:
Yeah good point, I think as long as MECs are doing integration reviews we'll be OK